<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>GenerateGraphWidget</class>
 <widget class="QWidget" name="GenerateGraphWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>354</width>
    <height>423</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Generate Graph</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="0" column="2">
    <layout class="QHBoxLayout" name="advancedLayout">
     <item>
      <widget class="KComboBox" name="comboGraphGenerator">
       <property name="toolTip">
        <string>Select the graph generator.</string>
       </property>
       <item>
        <property name="text">
         <string>Mesh Graph</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Star Graph</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Circle Graph</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Random Graph</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Erd√∂s-Renyi Graph</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Random Tree Graph</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Random Dag Graph</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Path Graph</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Complete Graph</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Complete Bipartite</string>
        </property>
       </item>
      </widget>
     </item>
     <item alignment="Qt::AlignRight">
      <widget class="QPushButton" name="buttonShowAdvanced">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="maximumSize">
        <size>
         <width>24</width>
         <height>24</height>
        </size>
       </property>
       <property name="toolTip">
        <string>Show advanced settings.</string>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="checkable">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="1" column="2">
    <widget class="QLineEdit" name="identifier">
     <property name="toolTip">
      <string>Set the unique identifier of the generated graph.</string>
     </property>
     <property name="text">
      <string>Graph</string>
     </property>
    </widget>
   </item>
   <item row="3" column="0">
    <widget class="QLabel" name="labelNodeType">
     <property name="toolTip">
      <string>The node type to create the nodes of the graph with</string>
     </property>
     <property name="text">
      <string>Node type</string>
     </property>
    </widget>
   </item>
   <item row="1" column="0">
    <widget class="QLabel" name="labelIdentifier">
     <property name="toolTip">
      <string>The identifier of the created graph (used for scripting)</string>
     </property>
     <property name="text">
      <string>Identifier</string>
     </property>
    </widget>
   </item>
   <item row="6" column="0" colspan="3">
    <widget class="QStackedWidget" name="stackedWidget">
     <property name="currentIndex">
      <number>3</number>
     </property>
     <widget class="QWidget" name="page_mesh">
      <layout class="QGridLayout" name="gridLayout_2">
       <item row="0" column="0">
        <layout class="QFormLayout" name="formLayoutMesh">
         <property name="fieldGrowthPolicy">
          <enum>QFormLayout::ExpandingFieldsGrow</enum>
         </property>
         <item row="0" column="0">
          <widget class="QLabel" name="label_columns">
           <property name="text">
            <string>Number of Columns:</string>
           </property>
           <property name="buddy">
            <cstring>meshColumns</cstring>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QSpinBox" name="meshColumns">
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="value">
            <number>5</number>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_rows">
           <property name="text">
            <string>Number of Rows:</string>
           </property>
           <property name="buddy">
            <cstring>meshRows</cstring>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QSpinBox" name="meshRows">
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="value">
            <number>5</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="page_star">
      <layout class="QGridLayout" name="gridLayout_3">
       <item row="0" column="0">
        <layout class="QFormLayout" name="formLayoutStar">
         <item row="0" column="0">
          <widget class="QLabel" name="label_numberSatellite">
           <property name="text">
            <string>Satellite Nodes:</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QSpinBox" name="starSatelliteNodes">
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="maximum">
            <number>999</number>
           </property>
           <property name="value">
            <number>10</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="page_circle">
      <layout class="QGridLayout" name="gridLayout_4">
       <item row="0" column="0">
        <layout class="QFormLayout" name="formLayoutCircle">
         <item row="0" column="0">
          <widget class="QLabel" name="label_numberCircle">
           <property name="text">
            <string>Number of Nodes:</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QSpinBox" name="circleNodes">
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="maximum">
            <number>999</number>
           </property>
           <property name="value">
            <number>10</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="page_random">
      <layout class="QGridLayout" name="gridLayout_5">
       <item row="0" column="0">
        <layout class="QFormLayout" name="formLayoutRandom">
         <item row="0" column="0">
          <widget class="QLabel" name="label_randomNodes">
           <property name="text">
            <string>Nodes:</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QSpinBox" name="randomNodes">
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="maximum">
            <number>999</number>
           </property>
           <property name="value">
            <number>10</number>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_randomEdges">
           <property name="text">
            <string>Edges:</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QSpinBox" name="randomEdges">
           <property name="value">
            <number>20</number>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="label_randomSelfEdges">
           <property name="text">
            <string>Allow self-edges:</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="label_randomGeneratorSeed">
           <property name="text">
            <string>Generator Seed:</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QSpinBox" name="randomGeneratorSeed">
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="maximum">
            <number>999999</number>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QCheckBox" name="randomAllowSelfedges">
           <property name="text">
            <string/>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="page_gnp">
      <layout class="QGridLayout" name="gridLayout_6">
       <item row="0" column="0">
        <layout class="QFormLayout" name="formLayoutGNP">
         <item row="0" column="0">
          <widget class="QLabel" name="label_GNPNodes">
           <property name="text">
            <string>Nodes (n):</string>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="label_GNPSelfEdges">
           <property name="text">
            <string>Allow self-edges:</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QSpinBox" name="GNPNodes">
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="value">
            <number>10</number>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_GNPEdgeProbability">
           <property name="text">
            <string>Edge Probability (p):</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="label_GNPGeneratorSeed">
           <property name="text">
            <string>Generator Seed:</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QSpinBox" name="GNPGeneratorSeed">
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="maximum">
            <number>999999</number>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QDoubleSpinBox" name="GNPEdgeProbability">
           <property name="decimals">
            <number>3</number>
           </property>
           <property name="minimum">
            <double>0.001000000000000</double>
           </property>
           <property name="maximum">
            <double>1.000000000000000</double>
           </property>
           <property name="value">
            <double>0.250000000000000</double>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QCheckBox" name="GNPAllowSelfedges">
           <property name="text">
            <string/>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="page_random_tree">
      <layout class="QGridLayout" name="gridLayout_8">
       <item row="0" column="0">
        <layout class="QFormLayout" name="formLayoutRandomTree">
         <item row="0" column="0">
          <widget class="QLabel" name="label_randomTreeNodes">
           <property name="text">
            <string>Nodes:</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QSpinBox" name="randomTreeNodes">
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="value">
            <number>10</number>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_randomTreeGeneratorSeed">
           <property name="text">
            <string>Generator Seed:</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QSpinBox" name="randomTreeGeneratorSeed">
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="maximum">
            <number>999999</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="page_random_dag">
      <layout class="QGridLayout" name="gridLayout_7">
       <item row="0" column="0">
        <layout class="QFormLayout" name="formLayoutRandomDag">
         <item row="0" column="0">
          <widget class="QLabel" name="label_numberNodes">
           <property name="text">
            <string>Number of Nodes</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QSpinBox" name="randomDagNumberOfNodes">
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="maximum">
            <number>999</number>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_edgeProbability">
           <property name="text">
            <string>Edge Probability</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QDoubleSpinBox" name="randomDagEdgeProbability">
           <property name="maximum">
            <double>1.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.100000000000000</double>
           </property>
           <property name="value">
            <double>0.500000000000000</double>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="label_dagGeneratorSeed">
           <property name="text">
            <string>Generator Seed</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QSpinBox" name="dagGeneratorSeed">
           <property name="enabled">
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="page_path">
      <layout class="QGridLayout" name="gridLayout_11">
       <item row="0" column="0">
        <layout class="QFormLayout" name="formLayoutPath">
         <item row="0" column="0">
          <widget class="QLabel" name="labe_pathNodes">
           <property name="text">
            <string>Number of Nodes:</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QSpinBox" name="pathNodes">
           <property name="value">
            <number>4</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="page_complete">
      <layout class="QGridLayout" name="gridLayout_10">
       <item row="0" column="0">
        <layout class="QFormLayout" name="formLayoutComplete">
         <item row="0" column="0">
          <widget class="QLabel" name="label_completeNodes">
           <property name="text">
            <string>Number of Nodes:</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QSpinBox" name="completeNodes">
           <property name="value">
            <number>5</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="page_complete_bipartite">
      <layout class="QGridLayout" name="gridLayout_12">
       <item row="0" column="0">
        <layout class="QFormLayout" name="formLayoutCompleteBipartite">
         <item row="0" column="0">
          <widget class="QLabel" name="label_completeBipartiteNodesLeft">
           <property name="text">
            <string>Left Set Nodes:</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QSpinBox" name="completeBipartiteNodesLeft">
           <property name="value">
            <number>4</number>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_completeBipartiteNodesRight">
           <property name="text">
            <string>Right Set Nodes:</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QSpinBox" name="completeBipartiteNodesRight">
           <property name="value">
            <number>4</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item row="7" column="2">
    <widget class="QDialogButtonBox" name="buttons">
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
   <item row="2" column="2">
    <widget class="KComboBox" name="edgeTypeSelector">
     <property name="toolTip">
      <string>Select the edge type for connections of the generated graph.</string>
     </property>
    </widget>
   </item>
   <item row="5" column="0" colspan="3">
    <widget class="Line" name="line">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
   </item>
   <item row="2" column="0">
    <widget class="QLabel" name="labelEdgeType">
     <property name="toolTip">
      <string>The edge type to create the edges of the graph with</string>
     </property>
     <property name="text">
      <string>Edge type</string>
     </property>
    </widget>
   </item>
   <item row="0" column="0">
    <widget class="QLabel" name="labelGenerator">
     <property name="text">
      <string>Graph Generator</string>
     </property>
    </widget>
   </item>
   <item row="3" column="2">
    <widget class="KComboBox" name="nodeTypeSelector">
     <property name="toolTip">
      <string>Select the node type for node elements of the generated graph.</string>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>KComboBox</class>
   <extends>QComboBox</extends>
   <header>kcombobox.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>buttonShowAdvanced</sender>
   <signal>toggled(bool)</signal>
   <receiver>label_randomGeneratorSeed</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>242</x>
     <y>21</y>
    </hint>
    <hint type="destinationlabel">
     <x>46</x>
     <y>158</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonShowAdvanced</sender>
   <signal>toggled(bool)</signal>
   <receiver>randomGeneratorSeed</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>242</x>
     <y>21</y>
    </hint>
    <hint type="destinationlabel">
     <x>98</x>
     <y>158</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonShowAdvanced</sender>
   <signal>toggled(bool)</signal>
   <receiver>label_GNPGeneratorSeed</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>242</x>
     <y>21</y>
    </hint>
    <hint type="destinationlabel">
     <x>18</x>
     <y>158</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonShowAdvanced</sender>
   <signal>toggled(bool)</signal>
   <receiver>GNPGeneratorSeed</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>242</x>
     <y>21</y>
    </hint>
    <hint type="destinationlabel">
     <x>83</x>
     <y>158</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonShowAdvanced</sender>
   <signal>toggled(bool)</signal>
   <receiver>label_randomTreeGeneratorSeed</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>242</x>
     <y>21</y>
    </hint>
    <hint type="destinationlabel">
     <x>46</x>
     <y>159</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonShowAdvanced</sender>
   <signal>toggled(bool)</signal>
   <receiver>randomTreeGeneratorSeed</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>242</x>
     <y>21</y>
    </hint>
    <hint type="destinationlabel">
     <x>98</x>
     <y>159</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>comboGraphGenerator</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>stackedWidget</receiver>
   <slot>setCurrentIndex(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>169</x>
     <y>16</y>
    </hint>
    <hint type="destinationlabel">
     <x>146</x>
     <y>194</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonShowAdvanced</sender>
   <signal>toggled(bool)</signal>
   <receiver>label_dagGeneratorSeed</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>331</x>
     <y>23</y>
    </hint>
    <hint type="destinationlabel">
     <x>70</x>
     <y>249</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonShowAdvanced</sender>
   <signal>toggled(bool)</signal>
   <receiver>dagGeneratorSeed</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>331</x>
     <y>23</y>
    </hint>
    <hint type="destinationlabel">
     <x>238</x>
     <y>249</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
